chore(docs): Release Notes for 5.13 (#38759)
gatsbyjs/gatsby: docs/docs/reference/release-notes/v5.13/index.md

---

> date: "2023-12-@TODO"

Author: need to update when released

Author: set it to Monday, to not publish stable toward my EoD on Friday

---
date: "2023-12-@TODO"
version: "5.13.0"
title: "v5.13 Release Notes"
---

Welcome to `gatsby@5.13.0` release (December 2023 #1)

Key highlight of this release:

- [Custom Image and File CDN URL generators](#custom-image-and-file-cdn-url-generators)

Also check out [notable bugfixes](#notable-bugfixes--improvements).

**Bleeding Edge:** Want to try new features as soon as possible? Install `gatsby@next` and let us know if you have any [issues](https://github.com/gatsbyjs/gatsby/issues).

> [Previous release notes](/docs/reference/release-notes/v5.12)

Author: links starting with `/docs` will not work in github - release notes are meant to be rendered on gatsbyjs.com site where those relative links would work (https://www.gatsbyjs.com/docs/reference/release-notes/v5.12/ is example of release notes being rendered there and "previous release notes" on that also use link like that https://github.com/gatsbyjs/gatsby/blob/eed07f3da59cd3d94b9e8c0b12aa1570f8a99a60/docs/docs/reference/release-notes/v5.12/index.md#L17 and in rendered version it works correctly)

[Full changelog][full-changelog]

---

## Custom Image and File CDN URL generators

> [`gatsby@4.10.0` introduced Image CDN](/docs/reference/release-notes/v4.10/#image-cdn) capability for Gatsby users. For best build performance it requires deployed site to handle image transformations at request time instead of at build time. Until now, handling those requests required supporting Gatsby specific URL format. With this release we're introducing support for custom URL generators that allow you to use your own URL format making it easier to integrate with existing site deployment platforms that offer Image CDN capabilities or using self managed Image CDNs by providing a way to generate URLs that are already in format expected by your CDN.

Author: links starting with `/docs` will not work in github - release notes are meant to be rendered on gatsbyjs.com site where those relative links would work (https://www.gatsbyjs.com/docs/reference/release-notes/v5.12/ is example of release notes being rendered there)

> [`gatsby@4.10.0` introduced Image CDN](/docs/reference/release-notes/v4.10/#image-cdn) capability for Gatsby users. For best build performance it requires deployed site to handle image transformations at request time instead of at build time. Until now, handling those requests required supporting Gatsby specific URL format. With this release we're introducing support for custom URL generators that allow you to use your own URL format making it easier to integrate with existing site deployment platforms that offer Image CDN capabilities or using self managed Image CDNs by providing a way to generate URLs that are already in format expected by your CDN.

Reviewer: ```suggestion
[`gatsby@4.10.0` introduced Image CDN](/docs/reference/release-notes/v4.10/#image-cdn) capability for Gatsby users. For best build performance, it requires a deployed site to handle image transformations at request time instead of at build time. Until now, handling those requests required supporting the specific URL format produced by Gatsby. With this release, we're introducing support for custom URL generators that allow you to use your own URL format. This makes it easier to integrate with existing site deployment platforms that offer Image CDN capabilities or to use self-managed Image CDNs by providing a way to generate URLs that are already in format expected by your CDN.
```

> Adapter API now offers ability to provide custom URL generators modules for both Image CDN (`gatsbyImage`) and File CDN (`publicUrl`). `gatsby-adapter-netlify@1.1.0` is optioanlly using this feature to make use of [Netlify's Image CDN](https://docs.netlify.com/image-cdn/overview/). Be sure to check [Netlify Adapter documentation](https://www.npmjs.com/package/gatsby-adapter-netlify#imagecdn) for more information how to enable and configure it.

Author: adapter npm readme link is not yet because it wasn't released to stable YET, but https://www.npmjs.com/package/gatsby-adapter-netlify/v/next#imagecdn (readme for `@next`) shows what it would link to once released to stable

> Adapter API now offers ability to provide custom URL generators modules for both Image CDN (`gatsbyImage`) and File CDN (`publicUrl`). `gatsby-adapter-netlify@1.1.0` is optioanlly using this feature to make use of [Netlify's Image CDN](https://docs.netlify.com/image-cdn/overview/). Be sure to check [Netlify Adapter documentation](https://www.npmjs.com/package/gatsby-adapter-netlify#imagecdn) for more information how to enable and configure it.

Reviewer: ```suggestion
Adapter API now offers ability to provide custom URL generators modules for both Image CDN (`gatsbyImage`) and File CDN (`publicUrl`). `gatsby-adapter-netlify@1.1.0` is optionally using this feature to make use of [Netlify's Image CDN](https://docs.netlify.com/image-cdn/overview/). Be sure to check [Netlify Adapter documentation](https://www.npmjs.com/package/gatsby-adapter-netlify#imagecdn) for more information on how to enable and configure it.
```

> Image and File CDN are using remote URL for source files. It is recommended to only allow certain remote URLs to be used to prevent abuse. This verions also introduced a way for source plugins that support Image and File CDN to declare list of allowed remote URLs with newly added [`addRemoteFileAllowedUrl` action](/docs/reference/config-files/actions/#addRemoteFileAllowedUrl). Adapters will receive list of allowed URLs to automatically configure or guide users on how to confiure their deployment to allow those remote URLs.

Reviewer: ```suggestion
Image and File CDN now use remote URLs for source files. It is recommended to only allow certain remote URLs to be used to prevent abuse. This version also introduced a way for source plugins that support Image and File CDN to declare a list of allowed remote URLs with the newly added [`addRemoteFileAllowedUrl` action](/docs/reference/config-files/actions/#addRemoteFileAllowedUrl). Adapters will receive a list of allowed URLs to automatically configure or guide users on how to configure their deployment to allow those remote URLs.
```

> Latest versions of `gatsby-source-contentful` (`8.13.0`), `gatsby-source-drupal` (`6.13.0`) and `gatsby-source-wordpress` (`7.13.0`) will use this new action to declare allowed remote URLs. Latest version of `gatsby-adapter-netlify` (`1.1.0`) will use this information to help configure Netlify's Image CDN to allow those remote URLs by providing informative message in case your `netlify.toml` doesn't allow required remote URLs.

Reviewer: ```suggestion
The latest versions of Gatsby plugins - `gatsby-source-contentful` (`v8.13.0`), `gatsby-source-drupal` (`v6.13.0`), and `gatsby-source-wordpress` (`v7.13.0`) - now incorporate a new action to specify allowed remote URLs. Additionally, `gatsby-adapter-netlify` (`v1.1.0`) leverages this update to assist in configuring Netlify's Image CDN. It ensures these remote URLs are permitted and provides informative messages if your `netlify.toml` file requires adjustments to support these URLs.
```

> If you are interested in using custom URL generators in your adapter, check source code of `gatsby-adapter-netlify` for reference implementation:

Reviewer: ```suggestion
If you are interested in using custom URL generators in your adapter, check the source code of `gatsby-adapter-netlify` for a reference implementation:
```
